### PR TITLE
Constellation documentation comments

### PR DESCRIPTION
**Pull request by asajeffrey**
Rewrote the documentation comments for `constellation.rs`.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because it's just documentation

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14698)

### CODE REVIEW

**Comment by asajeffrey** [comment]
cc @Ms2ger. r? @ConnorGBrewster

**Comment by asajeffrey** [comment]
Going through tidying this up generated a bunch of issues: #14690, #14691, #14692, #14693, #14694, and #14696.

File: components/constellation/constellation.rs [hunk 1]
```diff
@@ -191,62 +277,77 @@ pub struct Constellation<Message, LTF, STF> {
...
-impl FrameState {
-    fn new(pipeline_id: PipelineId, frame_id: FrameId) -> FrameState {
-        FrameState {
-            instant: Instant::now(),
-            pipeline_id: pipeline_id,
-            frame_id: frame_id,
-        }
-    }
+    /// Whether the constellation supports the clipboard.
+    /// TODO: this field is not used, remove it?
```
Comment (ConnorGBrewster) [inline components/constellation/constellation.rs]: This is due to #7578 (PR #12507). I suppose we could remove it and re-add it when clipboard support is added again, or just leave it for now.

File: components/constellation/constellation.rs [hunk 2]
```diff
@@ -256,36 +357,84 @@ impl Frame {
...
+            frame_id: frame_id,
+        }
+    }
+}
+
 /// Represents a pending change in the frame tree, that will be applied
 /// once the new pipeline has loaded and completed initial layout / paint.
 struct FrameChange {
+    /// The frame to change.
+
```
Comment (ConnorGBrewster) [inline components/constellation/constellation.rs]: nit: remove this line and add a new line below `frame_id`

File: components/constellation/constellation.rs [hunk 3]
```diff
@@ -684,6 +877,8 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
+    /// The joint session future is the merge of the session future of every
+    /// frame in the frame tree, sorted chronologically.
```
Comment (ConnorGBrewster) [inline components/constellation/constellation.rs]: Reverse chronologically?

**Review by cbrewster — APPROVED** [review]
Looks good other than a couple nits.

File: components/constellation/constellation.rs (hunk 1 above)
Comment (asajeffrey) [inline components/constellation/constellation.rs]: Just leave it I think.

File: components/constellation/constellation.rs (hunk 2 above)
Comment (asajeffrey) [inline components/constellation/constellation.rs]: Oops. Done.

File: components/constellation/constellation.rs (hunk 3 above)
Comment (asajeffrey) [inline components/constellation/constellation.rs]: Done.

**Comment by asajeffrey** [comment]
Squashed. @bors-servo r=ConnorGBrewster

**Comment by asajeffrey** [comment]
@bors-servo retry

- CI weirdness

**Comment by asajeffrey** [comment]
@jdm says this is https://github.com/servo/saltfs/issues/565

**Comment by asajeffrey** [comment]
YOLO @bors-servo retry

File: components/constellation/constellation.rs [hunk 4]
```diff
@@ -191,62 +277,77 @@ pub struct Constellation<Message, LTF, STF> {
...
-        }
-    }
+    /// Whether the constellation supports the clipboard.
+    /// TODO: this field is not used, remove it?
+    pub supports_clipboard: bool,
 }
 
-/// Stores the navigation context for a single frame in the frame tree.
+/// A frame in the frame tree.
+/// Each frame is the constrellation's view of a browsing context.
```
Comment (Ms2ger) [inline components/constellation/constellation.rs]: constrellation

Comment (asajeffrey) [inline components/constellation/constellation.rs]: Oops.

**Pull request merged**